src/ray/util/logging.cc

```diff
@@ -189,25 +189,20 @@ void RayLog::StartRayLog(const std::string &app_name, RayLogLevel severity_thres
         app_name_without_path = app_file_name;
       }
     }
-    char buffer[80];
-    time_t rawtime;
-    time(&rawtime);
-#ifdef _WIN32
-    int pid = _getpid();
-#else
-    pid_t pid = getpid();
-#endif
-    strftime(buffer, sizeof(buffer), "%Y%m%d-%H%M%S", localtime(&rawtime));
-    std::string path = dir_ends_with_slash + app_name_without_path + "." + buffer + "." +
-                       std::to_string(pid) + ".log";
-    stream_logger_singleton.out_.rdbuf()->pubsetbuf(0, 0);
-    stream_logger_singleton.out_.open(path.c_str(),
-                                      std::ios_base::app | std::ios_base::binary);
-  }
-  for (int lvl = 0; lvl < NUM_SEVERITIES; ++lvl) {
-    google::base::SetLogger(lvl, &stream_logger_singleton);
+    app_name_without_path += ".";
+    google::SetLogFilenameExtension(app_name_without_path.c_str());
+    int level = GetMappedSeverity(static_cast<RayLogLevel>(severity_threshold_));
+    google::SetLogDestination(level, dir_ends_with_slash.c_str());
```

Comment:
Does it mean we use the default logger provided by GLOG to write logs to a file if `log_dir` is provided, and we only use `StreamLogger` when we want to print logs to stdout?

Maybe I missed something, but it seems to me that you are reverting the work @mehrdadn has recently done.

Reply:
Did you mean we should skip all log outputs if log_dir is empty? I'm not sure it could be that. Any suggestion about that? @mehrdadn

Reply:
I find this fix PR https://github.com/ray-project/ray/pull/9633 and also keep the original style that error logs will be printed in stderr.

Reply:
The only work I did, overall, was to put log files in the log directory. Everything else was just fixing bugs that arose from my initial attempt to do that (such as the fact that different severities ended up in different log files). As long as you're putting log files in the log directory and the log files look reasonable, I don't think you're undoing anything I did.